### PR TITLE
Add EnableSSE3_4 as clr configuration knob in the documentation

### LINKED ISSUE #9659: Add EnableSSE3_4 as clr configuration knob in the documentation
In the clr configuration knob documentation at https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/clr-configuration-knobs.md, EnableAVX is documented, however, EnableSSE3_4 is not documented, we should add EnableSSE3_4 to the configuration document to help other developers who want to use this configuration knob when prototyping and testing.

The related code is at: https://github.com/dotnet/coreclr/blob/master/src/jit/jitconfigvalues.h#L198-#L209, both EnableSSE3_4 and EanbleAVX are already there.

We can provide a related PR for this, @fiigii PTAL

### PR DESCRIPTION
**Pull request by fiigii**
Fix #9659 

This PR modifies `clr-complus-conf-docgen.sh` to generate the CLR configuration knob doc from `jitconfigvalues.h` and `clrconfigvalues.h` both, which brings new JIT configuration knob into the documentation by macro `JIT_CONFIG_***_INFO`.

### CODE REVIEW

**Comment by fiigii** [comment]
@sivarv @RussKeldorph PTAL

File: src/inc/clrconfigvalues.h [hunk 1]
```diff
@@ -558,12 +558,15 @@ CONFIG_DWORD_INFO_EX(INTERNAL_JitVNMapSelLimit, W("JitVNMapSelLimit"), 0, "If no
...
 #define EXTERNAL_FeatureSIMD_Default 1
+#define EXTERNAL_JitEnableSSE3_4_Default 1
 #define EXTERNAL_JitEnableAVX_Default 1
 #else // !defined(_TARGET_AMD64_) && !defined(_TARGET_X86_)
 #define EXTERNAL_FeatureSIMD_Default 0
+#define EXTERNAL_JitEnableSSE3_4_Default 0
 #define EXTERNAL_JitEnableAVX_Default 0
 #endif // !defined(_TARGET_AMD64_) && !defined(_TARGET_X86_)
 RETAIL_CONFIG_DWORD_INFO_EX(EXTERNAL_FeatureSIMD, W("FeatureSIMD"), EXTERNAL_FeatureSIMD_Default, "Enable SIMD support with companion SIMDVector.dll", CLRConfig::REGUTIL_default)
+RETAIL_CONFIG_DWORD_INFO_EX(EXTERNAL_EnableSSE3_4, W("EnableSSE3_4"), EXTERNAL_JitEnableSSE3_4_Default, "Enable SSE3, SSSE3, and SSE4 instruction set for wide operations as default", CLRConfig::REGUTIL_default)
```
Comment (sivarv) [inline src/inc/clrconfigvalues.h]: >RETAIL_CONFIG_DWORD_INFO_EX(EXTERNAL_EnableSSE3_4, W("EnableSSE3_4"), EXTERNAL_JitEnableSSE3_4_Defa [](start = 0, length = 99)

We no longer need to add new jit specific config knobs in clrconfigvalues.h.  The right file is jit\jitconfigvalues.h

Btw, we already have this config knob in jit\jitconfigvalues.h

Comment (fiigii) [inline src/inc/clrconfigvalues.h]: @sivarv I see, but [clr-complus-conf-docgen.sh](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/clr-complus-conf-docgen.sh) reads `clrconfigvalues.h` to generate the doc. Do you think we should change `clr-complus-conf-docgen.sh` to generate the doc from `clrconfigvalues.h` and `jitconfigvalues.h` both, and remove jit specific config knobs from `clrconfigvalues.h`?

Comment (sivarv) [inline src/inc/clrconfigvalues.h]: @dotnet/jit-contrib 

Initially only clrconfigvalues.h existed containing both clr-vm and jit specific config knobs.  With jitconfigvalues.h we have separated jit-specific knobs into that file but still kept already existing ones in clrconfigvalues.h because Desktop needs it.  The new jit-specific config values are to be added to jitconfigvalues.h only.  So to generate documentation, I would suggest to take the union of clrconfigvalues.h and jitconfigvalues.h.

Comment (fiigii) [inline src/inc/clrconfigvalues.h]: Got it. Will do.

**Comment by fiigii** [comment]
@sivarv Done.

**Comment by danmosemsft** [comment]
@sivarv can we merge this old PR

**Comment by sivarv** [comment]
Looks good to me.
@RussKeldorph - an additional pair of eyes on scripting changes would be good before merging this.

**Comment by RussKeldorph** [comment]
@BruceForstall PTAL and merge if good.

File: src/jit/jitconfigvalues.h [hunk 2]
```diff
@@ -197,6 +210,7 @@ CONFIG_INTEGER(AltJitAssertOnNYI, W("AltJitAssertOnNYI"), 1) // Controls the Alt
 
 #if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
 CONFIG_INTEGER(EnableSSE3_4, W("EnableSSE3_4"), 1) // Enable SSE3, SSSE3, SSE 4.1 and 4.2 instruction set as default
+JIT_CONFIG_DWORD_INFO(EnableSSE3_4, W("EnableSSE3_4"), 1, "Enable SSE3, SSSE3, and SSE4 instruction set as default")
```
Comment (BruceForstall) [inline src/jit/jitconfigvalues.h]: Why is this added? This doesn't look right.

Comment (fiigii) [inline src/jit/jitconfigvalues.h]: Macro `JIT_CONFIG_DWORD_INFO` is used to generate the doc. Because most of the JIT config values are already included in `clrconfigvalues.h`, here we only need to generate doc for new JIT config values that are not in `clrconfigvalues.h`, so I think a new macro is necessary.

Comment (BruceForstall) [inline src/jit/jitconfigvalues.h]: We do not want to be adding new macros (that duplicate other macros) just to create the doc. The doc creation tool needs to be able to take the union of the knobs in both places.
